runtime/caml/gc_stats.h

```diff
@@ -0,0 +1,82 @@
+/**************************************************************************/
+/*                                                                        */
+/*                                 OCaml                                  */
+/*                                                                        */
+/*             Gabriel Scherer, projet Partout, INRIA Saclay              */
+/*                                                                        */
+/*   Copyright 2022 Institut National de Recherche en Informatique et     */
+/*     en Automatique.                                                    */
+/*                                                                        */
+/*   All rights reserved.  This file is distributed under the terms of    */
+/*   the GNU Lesser General Public License version 2.1, with the          */
+/*   special exception on linking described in the file LICENSE.          */
+/*                                                                        */
+/**************************************************************************/
+
+#ifndef CAML_GC_STATS_H
+#define CAML_GC_STATS_H
+
+#ifdef CAML_INTERNALS
+
+#include "domain.h"
+
+/* The Gc module provides two kind of runtime statistics:
+   - Heap stats: statistics about heap memory, see shared_heap.c
+   - Allocation stats: statistics about past allocations and collections,
+     see major_gc.c
+
+   To avoid synchronization costs, each domain tracks its own statistics.
+   We never access the "live stats" of other domains running concurrently,
+   only their "sampled stats", a past version of the live stats saved
+   at a safepoint -- during stop-the-world minor collections.
```

Comment:
We also collect statistics at domain termination IIRC.
Though domain termination induce a minor collection so it may be redundant?

Reply:
I added an explicit mention, thanks.